### ISSUE TITLE
False positive "unused-import" for typing python 3.7.4

### ISSUE THREAD

**Issue by davidgiga1993** [PyCQA/pylint#3193]
python 3.7.4 on linux:
```python
from typing import Optional

class A:
    def __init__(self):
        self.field = None  # type: Optional[str]
```

```bash
pylint  Test.py
************* Module Test
...
Test.py:1:0: W0611: Unused Optional imported from typing (unused-import)

pylint --version
pylint 2.4.2
astroid 2.3.1
Python 3.7.4 (default, Jul 23 2019, 10:08:33)
[GCC 4.8.5]
```

Interestingly I don't get this issue on windows with python 3.7.3 or 3.7.4:
```
pylint --version
pylint 2.4.2
astroid 2.3.1
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 21:26:53) [MSC v.1916 32 bit (Intel)]
```

Any ideas what could cause this issue?

**Comment by PCManticore** [comment]
Can you check if you have `typed-ast` installed?

**Comment by davidgiga1993** [comment]
Yes typed_ast is installed on both systems in V1.4.0

**Comment by saf** [comment]
Isn't this a duplicate of #3191, or at least the same root cause?

**Comment by davidgiga1993** [comment]
Could be yes. Although it would raise the question why it's working on windows but not on linux

**Comment by PCManticore** [comment]
@davidgiga1993 I can't reproduce this issue on my computer with `typed-ast==1.4.0` and
```
pylint 2.4.3
astroid 2.3.3
Python 3.7.0 (default, Jul  4 2018, 10:51:54)
[Clang 8.1.0 (clang-802.0.42)]
```

I'm not sure what the culprit might be given you already have `typed-ast` installed. As a last resort, you can do a breakpoint in `leave_assign` in `pylint/checkers/variables.py` and check if the node that is received for that method has a `type_annotation` attribute properly set. If it has, there might be a bug somewhere else down the line, but if not, it most likely is an issue with `typed-ast` itself.
Sorry I can't be of more help but cannot currently reproduce it.

**Comment by rpdelaney** [comment]
I came here to report this issue that I observed in a module using `weakref`. Since it's built-in, hopefully this repro will work for you:

```python
from typing import Optional
import weakref


class foo:
    def __init__(self, thing):
        self.bar = thing  # type:Optional[weakref.ref[int]]
```

```
$ pylint --version && pylint foo.py
pylint 2.4.3
astroid 2.3.3
Python 3.7.4 (default, Oct  4 2019, 06:57:26) 
[GCC 9.2.0]
************* Module foo
foo.py:2:0: W0611: Unused import weakref (unused-import)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 8.00/10, +0.00)
```
